### ISSUE TITLE
Change order of `deletedEntities`

### ISSUE THREAD

**Issue by elbotho** [serlo/database-layer#257]
# How should it behave?

`deletedEntities` query should return nodes with a newer `dateOfDeletion` first.

**Comment by elbotho** [comment]
@MoeHome is this a question for me or for @hugotiburtino ?

**Comment by MoeHome** [comment]
@elbotho for you. The main reason why I ask, is that I want to be sure you don't want the ids 4 and 3 as result of the above example. Minor reason is that I'm interested why you want the ordering that way.

**Comment by elbotho** [comment]
The way the frontend consumes this data is quite simple.
We display a list of recently deleted Entities ordered by date of deletion with the newly deleted entities first.
But we implement a kind of pagination so we don't have to load all entries at once. But as a consumer of the API I never knowingly query "entities that have been deleted after…". I just query more (and therefore older) entities.

Example:
I query 5 entries (get the 5 newest) and then query 5 more (get the next) and so on.

I don't know how those are then requested from the db-layer by the API.

**Comment by kulla** [comment]
@MoeHome you can check it in https://api.serlo-staging.dev/___graphql -> Take the query

```graphql
{
  entity {
    deletedEntities(first: 3) {
      nodes {
        dateOfDeletion
        entity {
          id
        }
      }
    }
  }
}
```

It returns

```
{
  "data": {
    "entity": {
      "deletedEntities": {
        "nodes": [
          {
            "dateOfDeletion": "2014-03-10T13:26:52+01:00",
            "entity": {
              "id": 14809
            }
          },
          {
            "dateOfDeletion": "2014-03-11T12:11:51+01:00",
            "entity": {
              "id": 14779
            }
          },
          {
            "dateOfDeletion": "2014-03-13T15:16:01+01:00",
            "entity": {
              "id": 17635
            }
          }
        ]
      }
    }
  }
}
```

In the ordering entity 17635 needs to be before 14779 since it is not so long deleted as 14779

**Comment by kulla** [comment]
When you run 

```graphql
{
  entity {
    deletedEntities(first: 5) {
      nodes {
        dateOfDeletion
        entity {
          id
        }
      }
    }
  }
}
```

you get 

```json
{
  "data": {
    "entity": {
      "deletedEntities": {
        "nodes": [
          {
            "dateOfDeletion": "2014-03-10T13:26:52+01:00",
            "entity": {
              "id": 14809
            }
          },
          {
            "dateOfDeletion": "2014-03-11T12:11:51+01:00",
            "entity": {
              "id": 14779
            }
          },
          {
            "dateOfDeletion": "2014-03-13T15:16:01+01:00",
            "entity": {
              "id": 17635
            }
          },
          {
            "dateOfDeletion": "2014-03-13T17:13:05+01:00",
            "entity": {
              "id": 17740
            }
          },
          {
            "dateOfDeletion": "2014-03-14T09:38:55+01:00",
            "entity": {
              "id": 17768
            }
          }
        ]
      }
    }
  }
}
```

Comparing with the above example with `first: 3` you see that the list of deleted entities has the wrong ordering. The newest deleted entities shall be first.

**Comment by hugotiburtino** [comment]
The problem persists: in staging (DB Layer version 0.3.43) 
https://api.serlo-staging.dev/___graphql
```graphql
{
  entity {
    deletedEntities(/DAs) {
      nodes {
        dateOfDeletion
        entity {
          id
        }
      }
    }
  }
}
```
it doesn't show the last deleted entities.
You can test it after logging in in serlo-staging.dev and putting in serlo-staging.dev/___graphql:
```graphql
mutation($input: UuidSetStateInput!) {
  uuid {
  	setState(input: $input) {
      success
      query {
        uuid(id: 1517) { // put here the id of entity
          trashed
        }
      }
    }
  }
}

variable:
{
    "input": { 
      "id": 1517,  // same id as above
      "trashed": true 
    }
}
```

**Comment by elbotho** [comment]
(just courious: can't you reuse the legacy sql query?)

**Comment by hugotiburtino** [comment]
Because legacy gets all trashed and [only then builds the pagination](https://github.com/serlo/serlo.org/blob/e720f0888fc87b747b3a05b8bf0731a7228c1068/packages/public/server/src/module/Uuid/src/Manager/UuidManager.php#L152), while we are trying to implement the pagination also for fetching the DB, that is the root of this bug.
(To be honest, we have seen the legacy only after your comment 😛 )
But we are almost there and it is going to be fully tested...

**Comment by elbotho** [comment]
Nice, good luck 👍 

> Because legacy gets all trashed and [only then builds the pagination](https://github.com/serlo/serlo.org/blob/e720f0888fc87b747b3a05b8bf0731a7228c1068/packages/public/server/src/module/Uuid/src/Manager/UuidManager.php#L152)

this is probably also why it sometimes takes a long time or fails completely. 
For me a implementation like this would also be ok if we also add a cronjob that deletes entries for good after some time ✨
I don't think there are a whole lot of trashed entries during 1 month. 

We should probably do this anyway.

**Comment by hugotiburtino** [comment]
closed via #269
